### ISSUE TITLE
In-memory reading small datasets (mnist)

### ISSUE THREAD

**Issue by weiliu620** [tensorflow/models#508]
I guess the datasets in slim is defined to read from hard disk. I just tested the mnist data with slim and found it very slow compared to using native tensorflow code to read data in memory. 

Sorry to abuse this issue tracker. I just asked a question on stackoverflow but think here might be better place. Actually not an issue but more like a feature request: can I have a example of using data provider in slim to read data in memory, just for small dataset? 

The models are great tools. Thank you. 

`provider = slim.dataset_data_provider.DatasetDataProvider(train_set, num_readers = 10, shuffle = True)
        [image, label] = provider.get(['image', 'label'])
        images, _ = tf.train.batch([image, label], batch_size = batch_size, num_threads=10, capacity = 2*batch_size)`

**Comment by tatatodd** [comment]
@sguada might have some advice here.

**Comment by aselle** [comment]
Closing due to lack of recent activity and because it is indeed more of a stackoverflow question. It would great if you linked the stackoverflow question @weiliu620.
